File: 0128-longest-consecutive-sequence/0128-longest-consecutive-sequence.cpp
class Solution {
public:
    int longestConsecutive(vector<int>& nums) {
        if (nums.empty()) return 0;
        sort(nums.begin(), nums.end());
        
        int max_count = 0; 
        int count = 0;
        for(int i=0; i<nums.size()-1; i++) {
            if(nums[i+1] == nums[i]) continue;
            if(nums[i+1]-nums[i] == 1) {
                count+=1;
                max_count = max(max_count, count);
                continue;
            }

            max_count = max(max_count, count);
            count = 0;
        }

        return max_count+1;
    }
};

File: 0128-longest-consecutive-sequence/0128-longest-consecutive-sequence.py
class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:
        if not len(nums):
            return 0

        def checkSeq(numSet: set, n: int) -> int:
            nSeq = 1
            while n+1 in numSet:
                nSeq += 1
                n = n+1
            else:
                return nSeq

        numSet = set(nums)
        maxSeq = 1

        for n in numSet:
            if n-1 in numSet:
                continue
            else:
                maxSeq = max(checkSeq(numSet, n), maxSeq)

        return maxSeq